File: Data Structures & Algorithms/longest-consecutive-sequence/submission-4.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        s = set(nums)
        best = 0
        for n in nums:
            if n in s and n-1 not in s:
                case = [n]
                s.remove(n)
                while case[-1]+1 in s:
                    case.append(case[-1] + 1)
                    s.remove(case[-1])
                best = max(len(case), best)

        return best   
                

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-4.cs
public class Solution {
    public int LengthOfLongestSubstring(string s) {
        if (s.Length < 2) return s.Length;
        int best = 0;
        int l = 0;
        int r = 1;
        HashSet<char> set = new HashSet<char>();
        set.Add(s[l]);
        while (r < s.Length) {
            if (!set.Contains(s[r])) {
                set.Add(s[r]);
                r += 1;
            }
            else {
                while (set.Contains(s[r])) {
                    set.Remove(s[l]);
                    l += 1;
                }
            }
            best = Math.Max(best, r - l);
        }
        return best;
    }
}
